bindinfo/handle.go

```diff
@@ -123,17 +122,22 @@ func NewBindHandle(ctx sessionctx.Context) *BindHandle {
 func (h *BindHandle) Update(fullLoad bool) (err error) {
 	h.bindInfo.Lock()
 	lastUpdateTime := h.bindInfo.lastUpdateTime
+	updateTime := lastUpdateTime.String()
+	if fullLoad {
+		updateTime = "0000-00-00 00:00:00"
+	}
 
-	sql := "select original_sql, bind_sql, default_db, status, create_time, update_time, charset, collation, source from mysql.bind_info"
-	if !fullLoad {
-		sql += " where update_time > \"" + lastUpdateTime.String() + "\""
+	exec := h.sctx.Context.(sqlexec.RestrictedSQLExecutor)
+	stmt, err := exec.ParseWithParams(context.TODO(), `SELECT original_sql, bind_sql, default_db, status, create_time, update_time, charset, collation, source
+	FROM mysql.bind_info WHERE update_time > %? ORDER BY update_time`, updateTime)
```

Comment:
Will this condition be optimized?

Reply:
It changes it from a tablescan prior (with no filter) to a tablescan with a filter on `updateTime > 0000-00-00 00:00:00`.

It is not quite semantically identical, because there is one row in the table which is known to have the update_time of this (BuiltinPseudoSQL4BindLock). But you can see that is skipped when iterating through rows. Every other row should have a time > 0.